sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ColumnDefinition.scala

```diff
@@ -51,14 +56,22 @@ case class ColumnDefinition(
   }
 
   def toV2Column(statement: String): V2Column = {
+    val finalMetadata = if (identityColumnSpec.isDefined) {
```

Comment:
why do we need this while v2 column already has an explicit field for identity column spec?